### PR TITLE
[Prototype] Integrating PennyLane with error mitigation

### PR DESCRIPTION
**Pull request by trbromley**
This is a prototype of how error mitigation can be integrated into PennyLane using the `mitiq` package.

### CODE REVIEW

File: pennylane/transforms/mitigation.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        circuit (cirq.Circuit): input Cirq circuit
+        measurements (Iterable[.MeasurementProcess]): measurements to be appended to tape
+
+    Returns:
+        .QuantumTape: the corresponding :class:`~.QuantumTape`
+    """
+    try:
+        from mitiq.mitiq_qiskit import to_qiskit
+    except ImportError as e:
+        raise ImportError("The mitiq package is required") from e
```
Comment (anthayes92) [inline pennylane/transforms/mitigation.py:103]: It's possible that a user may have `mitiq` but not have `qiskit` and hit this error

Comment (anthayes92) [inline pennylane/transforms/mitigation.py:103]: It's possible that a user may have `mitiq` but not have `qiskit` and hit this error

File: pennylane/transforms/mitigation.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            r = results[indx : indx + shape]
+            mitigated_results.append(func(r))
+            indx += shape
+
+        return mitigated_results
+
+    dev.execute = execute
+    dev.batch_execute = batch_execute
+
+    return dev
```
Comment (Thenerdstation) [inline pennylane/transforms/mitigation.py:239]: 😍

File: pennylane/transforms/mitigation.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+            inverted_operation.inv()
+
+            inv_operation_map[key + Operation.string_for_inverse] = inverted_operation
+
+        cirq_operation_map = {**operation_map, **inv_operation_map}
+
+    return cirq_operation_map
+
+
+def _tape_to_cirq(tape):
```
Comment (josh146) [inline pennylane/transforms/mitigation.py:49]: I think this is a useful enough transform that it deserves not only to be public, but to be in its own module

Comment (trbromley) [reply pennylane/transforms/mitigation.py:49]: The cool thing is that we have a lot of these transformations already, just wrapped up in a device (e.g., this was from PennyLane-Cirq)

File: pennylane/transforms/mitigation.py (hunk 1 above)
Comment (trbromley) [reply pennylane/transforms/mitigation.py:107]: Thanks, fixed!

File: pennylane/transforms/mitigation.py (hunk 3 above)
Comment (josh146) [reply pennylane/transforms/mitigation.py:43]: yes, was thinking the same thing!

**Comment by nathanshammah** [comment]
Very cool!

**Comment by trbromley** [comment]
Closing as this was re-implemented in https://github.com/PennyLaneAI/pennylane/pull/1813

**Pull request closed without merging**